### DISCUSSION TITLE
Basic question regarding yarp::os::BufferedPort::read()!

### DISCUSSION THREAD

**Discussion by paliasgh** [robotology/community#683] [Q&A] [answered]
Hello and sorry for bothering you again,

I have a basic question about something I came across and I could not fix.  Let's say I have a `rf` module that runs every 0.1 seconds and reads from a port in every update through `messageRsPos = rsPos.read(false);` in `updateModule()`. Data in the port, however, is published much slower (let's say every 0.5 seconds). In this case, i.e., when no new data has come, `messageRsPos` returns a null pointer. Is this expected? How can I make it read the last message regardless (repeat the last message if there is no new one)? Thanks.

**Comment by pattacini** [comment, marked as answer]
Hi @paliasgh 

Never mind, we're here to help.

>In this case, i.e., when no new data has come, messageRsPos returns a null pointer. Is this expected?

Yep.

>How can I make it read the last message regardless (repeat the last message if there is no new one)?

An easy solution consists in latching the value in a class data member for example.
Something like the following:

```c++
messageRsPos = rsPos.read(false);
if (messageRsPos != nullptr) {
  messageRsPosLatch = *messageRsPos; 
}
```

In your code, you'll be using `messageRsPosLatch`, which gets updated only when a new message is retrieved via polling. Just make sure to initialize `messageRsPosLatch` with a value that is sensible for your application.

This should address your use case.
Instead, if you aim to query the `BufferedPort` to return the last data received, [`lastRead()`](https://yarp.it/latest//classyarp_1_1os_1_1BufferedPort.html#adb8b4fb15b894f24fa167ad7f65ed24e) should do the job.
